clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/consumer/PlaintextConsumerCommitTest.java

```diff
@@ -452,6 +452,34 @@ private void testPositionAndCommit(GroupProtocol groupProtocol) throws Interrupt
         }
     }
 
+    /**
+     * This is testing when closing the consumer but commit request has already been sent.
+     * During the closing, the consumer won't find the coordinator anymore.
+     */
+    @ClusterTest
+    public void testCommitAsyncFailsWhenCoordinatorUnavailableDuringClose() throws InterruptedException {
+        try (Producer<byte[], byte[]> producer = cluster.producer();
+             var consumer = createConsumer(GroupProtocol.CONSUMER, false)
+        ) {
+            sendRecords(producer, tp, 3, System.currentTimeMillis());
+            consumer.assign(List.of(tp));
+
+            var callback = new CountConsumerCommitCallback();
+
+            // Close the coordinator before committing because otherwise the commit will fail to find the coordinator.
+            cluster.brokerIds().forEach(cluster::shutdownBroker);
+
+            consumer.poll(Duration.ofMillis(500));
+            consumer.commitAsync(Map.of(tp1, new OffsetAndMetadata(1L)), callback);
+            consumer.close(CloseOptions.timeout(Duration.ofMillis(500)));
```

Comment:
Is it possible for `close()` to take more than 500 milliseconds? Should we time the method call and add an assert to ensure it's not more than, say, 1000 milliseconds?

Reply:
yes, +1 to add assert to ensure the elapsed time is not more than timeout